tests/vmi_tpm_test.go

```diff
@@ -0,0 +1,62 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package tests_test
+
+import (
+	expect "github.com/google/goexpect"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/console"
+)
+
+var _ = Describe("[sig-compute]vTPM", func() {
+	Context("with TPM VMI option enabled", func() {
+		It("should expose a functional emulated TPM", func() {
+			By("Creating a VMI with TPM enabled")
+			vmi := tests.NewRandomFedoraVMI()
+			vmi.Spec.Domain.Devices.TPM = &v1.TPMDevice{}
+			vmi = tests.RunVMIAndExpectLaunch(vmi, 60)
+
+			By("Logging in as root")
+			err := console.LoginToFedora(vmi)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Ensuring a TPM device is present")
+			Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
+				&expect.BSnd{S: "ls /dev/tpm*\n"},
+				&expect.BExp{R: "/dev/tpm0"},
+			}, 300)).To(Succeed(), "Could not find a TPM device")
+
+			By("Ensuring the TPM device is functional")
+			Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
+				&expect.BSnd{S: "tpm2_pcrread sha1:16\n"},
+				&expect.BExp{R: "0x0000000000000000000000000000000000000000"},
+				&expect.BSnd{S: "tpm2_pcrextend 16:sha1=FB17882585BBFE9C55733A6E46A265DDAEA6957A && echo 'do''ne'\n"},
```

Comment:
nit: `'do''ne'` -> `done`

Reply:
This is actually on purpose, to avoid `BExp` matching the command instead of the result.
`SafeExpectBatch` should prevent that from happening, but I'd rather be 100% sure that if the word "done" appears anywhere in the console, it's because the `pcrextend` succeeded.